mcos/optimizer.py

```diff
@@ -181,3 +183,112 @@ def _get_optimal_portfolio(self, cov: np.array, mu: np.array) -> np.array:
         w = np.dot(inv, mu)
         w /= np.dot(ones.T, w)
         return w.flatten()
+
+
+class HRPOptimizer(AbstractOptimizer):
+    """
+    Hierarichal Risk Parity Optimizer based on Dr. Marcos Lopez de Prado's paper 'Building Diversified Portfolios that
+     Outperform Out-of-Sample'
+    """
+
+    def allocate(self, mu: np.array, cov: np.array) -> np.array:
+        return self.get_hrp_weights(cov)
```

Comment:
Could we put all the content from `hierarchical_risk_parity_portfolio` into this `allocate` method? I'm not sure the `get_hrp_weights` and `hierarchical_risk_parity_portfolio` methods are needed.